python/ray/autoscaler/_private/gcp/node.py

```diff
@@ -196,15 +196,25 @@ class GCPTPUNode(GCPNode):
     def get_labels(self) -> dict:
         return self.get("labels", {})
 
-    def get_external_ip(self) -> str:
+    @property
+    def num_workers(self) -> int:
+        return len(self.get("networkEndpoints", [{}]))
+
+    def get_external_ips(self) -> List[str]:
+        return self.get("networkEndpoints", [{}])
+
+    def get_external_ip(self, worker_index: int = 0) -> str:
```

Comment:
Can you remind me why it makes sense to have a default worker index here? How are the return values of `get_external_ip()` and `get_internal_ip()` used by the caller when the index isn't specified?

Reply:
So node_provider has a function for `get_external_ip(node_ip)`. Since TPUs are the only node types (at least currently) that will have multiple worker indices, I figured it would be safe to set a default index to 0 which is esp. useful when we are using single TPU VMs.

We could remove the default value, but we would need to change the logic here: https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/_private/gcp/node_provider.py#L157 with a TPU check which is kind of ugly.

Reply:
Got it, that makes sense! The current way is fine in that case